server/src/main/java/org/elasticsearch/search/lookup/SourceLookup.java

```diff
@@ -32,24 +32,10 @@
 
 import static java.util.Collections.emptyMap;
 
-public class SourceLookup implements Map<String, Object> {
+public interface SourceLookup extends Map<String, Object> {
+    XContentType sourceContentType();
 
-    private LeafReader reader;
-    private CheckedBiConsumer<Integer, FieldsVisitor, IOException> fieldReader;
-
-    private int docId = -1;
-
-    private BytesReference sourceAsBytes;
-    private Map<String, Object> source;
-    private XContentType sourceContentType;
-
-    public XContentType sourceContentType() {
-        return sourceContentType;
-    }
-
-    public int docId() {
-        return docId;
-    }
+    int docId();
```

Comment:
While you are here, can you add javadoc for these? It'll really help whoever comes after. I know I often don't. But I should...